images/entrypoint.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# update ca certificates if they are injected with a volume mount
+sudo update-ca-certificates
```

Comment:
do we need to do this behind some ENV so we don't have to refresh the cert store every time?

Reply:
Not sure, it can delay startup slightly, but I think it won't be noticeable because the operations is pretty fast.
Inside the container, the operation took 0.360s so I guess it should be fine executing it each time.
What I did is:
`docker run --entrypoint bash -it {my image}`
Then executed `time sudo update-ca-certificates`

Reply:
@rentziass @Link- any thoughts? I have no idea to be honest. I think if someone wants to build the image, they would probably check to see this `Dockerfile` and the `entrypoint.sh`, but I have no idea tbh...

Reply:
another way is to run `update-ca-certificates (ubuntu/debian)` or `update-ca-trust(redhat/centos)` in the `run.sh` based on an environment variable passed in from ARC.
similar to `RUNNER_WAIT_FOR_DOCKER_IN_SECONDS` in the `run.sh`

Reply:
Wrote helper function to check for tools if env is exported. Please let me know what do you think about this approach :smile:

Reply:
@TingluoHuang @nikola-jokic - I definitely like the idea of triggering the update store based on the status of a flag. However, I'm failing to see how hiding that flag in `run.sh` would solve the problem of using other distributions? What if we add a couple of conditions to identify the base image use and decide (in addition to the flag) whether we should run this or not?

Reply:
To avoid extensive checks for the base image, I tried just seeing if cli utility is present in the image. That was the easiest way I could think of to avoid different formats for different distributions (I tried to see different formats of `/etc/os-release`, so those checks would be more difficult)

My understanding is that this env is just a signal to the `entrypoint.sh` to trigger ca update before the run.sh is called.
So if you don't have any mounts with certs, it would not be called

On the other hand, I would prefer to call it in every case, since anyone can mount custom certs as config maps for example, and the controller will be completely unaware of it.

The call is not too expensive, it executed in less than 1s when I tried it inside the image. But I don't know if that can become expensive on some nodes...